File: 기본 수학/1978/1978.py
M = 1001
answer = 0
prime = [1] * M
prime[0], prime[1] = 0, 0
for i in range(2, M):
   for j in range(i*2, M, i):
      prime[j] = 0

n = int(input())
arr = list(map(int, inputs().split()))
for i in arr:
   if prime[i]:
      answer += 1
print(answer)

File: 기본 수학/1978/1978.c
#include <stdio.h>
#define MAX 1001

int main()
{
   int n, i, j, t;
   int answer = 0;
   _Bool prime[MAX] = {1, 1, 0};
   for (i = 2; i < MAX; ++i)
      for (j = i * 2; j < MAX; j += i)
         prime[j] = 1;

   scanf("%d", &n);
   for (i = 0; i < n; ++i)
   {
      scanf("%d", &t);
      if (!prime[t])
         ++answer;
   }
   printf("%d", answer);

   return 0;
}